sync/sync.go

```diff
@@ -2,133 +2,176 @@
 package sync
 
 import (
+	"errors"
 	"github.com/golang/protobuf/proto"
 	"github.com/levenlabs/go-llog"
 	"github.com/levenlabs/saturn/config"
 	. "github.com/levenlabs/saturn/proto"
 	"net"
+	"strconv"
 	"time"
 )
 
 type MsgType byte
 
 const (
-	Report   MsgType = 2
+	Init     MsgType = 1
+	Report           = 2
 	Response         = 3
 )
 
 func SendReport(serverAddr *net.UDPAddr) {
-	now := time.Now()
-	req := makeReport(serverAddr.IP, now)
-	d, err := proto.Marshal(req)
+	//this is making a pretend job for a report to send that we can funnel through
+	//the normal flow of packets
+	c, err := net.DialUDP("udp", nil, serverAddr)
 	if err != nil {
-		llog.Error("marshal error in SendReport", llog.KV{"err": err})
+		llog.Error("dial error in SendReport", llog.KV{"err": err})
 		return
 	}
-	startTransaction(req.Trans, serverAddr, now, req.Name)
-	send(Report, serverAddr, d)
+	t := InitTransaction(serverAddr, c)
```

Comment:
Another package shouldn't have to know what go-routines to start up for a new transaction I don't think. Might be better to give transactions a `Start` method which can be called at this point, and inside `Start` is where these go-routines are created